### DISCUSSION TITLE
COMO CRIAR GRAFANA MIMIR NO ECS FARGATE

### DISCUSSION THREAD

**Discussion by macsmarte** [grafana/mimir#8226] [Development]
Bom dia pessoal, este é mo meu primeiro projeto de mimir e estou tentando subir algo simples que é a princípio grafana mimir no ECS Fargate, alguem pode me mostrar a configuração mais simples de dockerfile e mimir.yml para eu ten tar replicar aqui ?

Meu ero que estou tendo:

ts=2024-05-30T17:05:31.493276716Z caller=mimir.go:880 level=error msg="module failed" module=memberlist-kv err="starting module memberlist-kv: invalid service state: Failed, expected: Running, failure: service memberlist_kv failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"

**Comment by pstibrany** [comment]
> ts=2024-05-30T17:05:31.493276716Z caller=mimir.go:880 level=error msg="module failed" module=memberlist-kv err="starting module memberlist-kv: invalid service state: Failed, expected: Running, failure: service memberlist_kv failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"

This error is logged when Mimir can't find IP address to use for memberlist. You will need to pass explicit IP address in `-memberlist.advertise-addr` option or corresponding YAML option. This IP address will be told to other nodes, and other nodes should be able to talk to it.

**Reply by macsmarte** [reply]
@pstibrany, obrigado. mas pode ser o localhost?
e como ficaria no YAML esta instrução?

**Reply by pstibrany** [reply]
> @pstibrany, thank you. But could it be Localhost?and how would this instruction look in YAML?

It can be localhost, but then you're limited to running Mimir on single node only. In YAML it's 

```
memberlist:
    advertise_addr: 127.0.0.1
```

See https://grafana.com/docs/mimir/latest/configure/configuration-parameters/#memberlist for details.

**Reply by macsmarte** [reply]
obrigado @pstibrany como o storage é unicamente no s3, acredito que o autoscaling resolveria não?

**Reply by pstibrany** [reply]
Mimir is designed to be scaled horizontally -- by adding more Mimir instances to handle more load. Single-node Mimir is basically just a fancy Prometheus :)

**Reply by macsmarte** [reply]
legal, muito obrigado :)

**Comment by macsmarte** [comment]
fechando a discução como done :)
